packages/facility-server/__tests__/apiv1/User.test.js

```diff
@@ -340,108 +355,144 @@ describe('User', () => {
   });
 
   describe('User facility methods', () => {
-    let user = null;
     let superUser = null;
-    let noFacilityUser = null;
+    let linkedUser = null;
+    let unlinkedUser = null;
 
-    const validUserFacilities = [facility1, facility2];
-    const validUserFacilityIds = validUserFacilities.map((f) => f.id);
+    const nonSensitiveFacilities = [facility1, facility2, facility3];
+    const nonSensitiveFacilityIds = nonSensitiveFacilities.map(f => f.id);
 
-    beforeAll(async () => {
+    const linkedUserFacilities = [facility1, sensitiveFacility1];
+    const linkedUserFacilityIds = linkedUserFacilities.map(f => f.id);
+
+    const expectedCombinedFacilities = [...nonSensitiveFacilities, sensitiveFacility1];
+    const expectedCombinedFacilityIds = expectedCombinedFacilities.map(f => f.id);
+
+    const mockLoginFacilityPermission = async (user, hasPermission) => {
+      jest.spyOn(user, 'hasPermission').mockImplementation(() => hasPermission);
+    };
+
+    // Defaults for all of the tests in this block. We override as needed
+    beforeEach(async () => {
       await models.Setting.set('auth.restrictUsersToFacilities', true);
+      mockLoginFacilityPermission(linkedUser, false);
+    });
+
+    beforeAll(async () => {
       superUser = await models.User.create(
         createUser({
           role: 'admin',
         }),
       );
-      user = await models.User.create(
+
+      linkedUser = await models.User.create(
         createUser({
           role: 'practitioner',
         }),
       );
-      noFacilityUser = await models.User.create(
+
+      unlinkedUser = await models.User.create(
         createUser({
           role: 'practitioner',
         }),
       );
 
       await Promise.all(
-        validUserFacilities.map(async (facility) => {
+        linkedUserFacilities.map(async facility => {
           return await models.UserFacility.create({
             facilityId: facility.id,
-            userId: user.id,
+            userId: linkedUser.id,
           });
         }),
       );
 
-      await user.reload({ include: 'facilities' });
-    });
-
-    describe('checkCanAccessAllFacilities', () => {
-      it('should return true if superuser', async () => {
-        expect(await superUser.checkCanAccessAllFacilities()).toBe(true);
-      });
-      it('should return false if user without  "login", "Facility" permission', async () => {
-        expect(await user.checkCanAccessAllFacilities()).toBe(false);
-      });
+      await linkedUser.reload({ include: 'facilities' });
     });
 
     describe('allowedFacilities', () => {
       it('should get special "ALL" key when superuser', async () => {
         const superUserFacilities = await superUser.allowedFacilities();
         expect(superUserFacilities).toBe(CAN_ACCESS_ALL_FACILITIES);
       });
+
+      it('user with login permission should return all non-sensitive facilities plus linked facilities', async () => {
```

Comment:
Should read as a sentence starting with "it"
```suggestion
      it('should return all non-sensitive facilities plus linked facilities for a user with login permission', async () => {
```